Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -233,6 +233,47 @@ final class CacheAdvanceTests: XCTestCase {
         XCTAssertEqual(messages, Array(LorumIpsum.messages.dropFirst(2)) + [barelyLongerMessage])
     }
 
+    func test_append_inAnOverwritingCacheEvictsMinimumPossibleNumberOfMessages() throws {
+        let message: TestableMessage = "test-message"
+        let maxMessagesBeforeOverwriting = [message, message, message]
+        let cache = try createCache(
+            sizedToFit: maxMessagesBeforeOverwriting,
+            overwritesOldMessages: true)
+        for message in maxMessagesBeforeOverwriting {
+            try cache.append(message: message)
+        }
+
+        // All of our messages have been stored and our cache is full.
+
+        // The byte layout in the cache should now be as follows:
+        // [header][length|test-message][length|test-message][length|test-message]
+        //         ^ reading handle                                              ^ writing handle
```

Comment:
super nitty: Should the writing handle be pointing at the byte _past_ the end of the last message, rather than the last byte of the last message?

```suggestion
        //         ^ reading handle                                               ^ writing handle
```

Reply:
Yes! Good call-out! This is what I did further down: https://github.com/dfed/CacheAdvance/blob/7f2448dfc09b2ec1eff4b1ed7ee4800a95629e69/Tests/CacheAdvanceTests/CacheAdvanceTests.swift#L267-L268